front/utils/utils.ts

```diff
@@ -23,6 +23,31 @@ export function debounce<A = unknown, R = void>(
     });
 }
 
+
 export function formatNumber(number: number): string {
-  return number.toLocaleString('fr-FR');
+  let options = {
+    notation: "compact" as const,
+    currency: "EUR" as const,
+    style: "currency" as const,
+    maximumFractionDigits: 1,
+  };
+
+  let frenchformatter = new Intl.NumberFormat("fr-FR", options);
+  let FRformattedNumber = frenchformatter.format(number);
+
+  return FRformattedNumber.replace(/\s?€/, "€");
+}
+
+export function formatNumberFrench(number: number): string {
+  let options = {
+    currency: "EUR" as const,
+    style: "currency" as const,
+    maximumFractionDigits: 0,
+  };
+
+  let frenchformatter = new Intl.NumberFormat("fr-FR", options);
+  let FRformattedNumber = frenchformatter.format(number);
+
+  return FRformattedNumber;
```

Comment:
Utilise plutot des consts quand elles ne varient pas.

Tu peux creer deux fonctions a exporter, une pour formater un nombre en francais et une pour formater un nombre en euros ! Puis tu peux ajouter des options que tu peux ajouter directement ;)
```suggestion
function formatFrench(value: number, options?: Intl.NumberFormatOptions) {
  const formatter = new Intl.NumberFormat('fr-FR', options);
  const formattedNumber = formatter.format(value);

  return formattedNumber;
}

export function formatNumber(value: number, options?: Intl.NumberFormatOptions) {
  const defaultOptions = {
    maximumFractionDigits: 2,
    ...options,
  } as const;

  return formatFrench(value, defaultOptions);
}

export function formatPrice(value: number, options?: Intl.NumberFormatOptions): string {
  const defaultOptions = {
    style: 'currency',
    currency: 'EUR',
    maximumFractionDigits: 0,
    ...options,
  } as const;

  return formatFrench(value, defaultOptions);
}
```

Reply:
Oui, très juste